src/utils/__tests__/use-interval.test.ts

```diff
@@ -0,0 +1,47 @@
+import {renderHook} from '@testing-library/react-hooks';
+import {useInterval} from '../use-interval';
+
+describe('useInterval', () => {
+  let callback = jest.fn();
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+    jest.spyOn(global, 'setTimeout');
+  });
+
+  afterEach(() => {
+    callback.mockRestore();
+    jest.clearAllTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it('should invoke with specified delay', () => {
```

Comment:
```
it('should invoke with specified delay', () => {
  const delay = 300;
  const {result} = renderHook(() => useInterval(callback, delay));
  expect(callback).toHaveBeenCalledTimes(0);
  jest.advanceTimersByTime(delay);
  expect(result.current).toBeUndefined();
  expect(callback).toHaveBeenCalledTimes(1);
});
```

Reply:
These two tests seems to be working as intended with existing code.

Had troubles with `should invoke with specified delay and immediate if set` however. But looking more into this right now.

Reply:
Hmm, when I comment out 'should invoke with specified delay and immediate if set', then 'should invoke with specified delay' fails as well for me.

Reply:
Might be some setup/teardown timer issue making it non-deterministic behaviour.

Reply:
Testing some more it seems like faked testers in jest don't play nice with async code. I added some workarounds and it seems to be working as expected now. Might look into adding some more test cases soon.

Reply:
Very nice :+1: